### PR TITLE
Use mem_pool replace arena first step

### PR DESCRIPTION
**Pull request by kangkaisen**
For https://github.com/apache/incubator-doris/issues/1898

Why use mem_pool replace arena?

1. **The mem_pool and arena is the same essentially**
2. **mem_pool is more powerful than arena**
3. **mem_pool is widely used**

We should use mem_pool replace arena step by step, and finally remove arena.

### CODE REVIEW

File: be/src/olap/memtable.h [hunk 1]
```diff
@@ -57,11 +57,12 @@ class MemTable {
...
 
     RowCursorComparator _row_comparator;
-    Arena _arena;
+    std::unique_ptr<MemTracker> _tracker;
+    std::unique_ptr<MemPool> _mem_pool;
     ObjectPool _agg_object_pool;
 
     typedef SkipList<char*, RowCursorComparator> Table;
-    char* _tuple_buf;
+    u_int8_t* _tuple_buf;
```
Comment (imay) [inline be/src/olap/memtable.h]:
```suggestion
    uint8_t* _tuple_buf;
```

File: be/src/olap/skiplist.h [hunk 2]
```diff
@@ -186,7 +182,7 @@ struct SkipList<Key,Comparator>::Node {
 template<typename Key, class Comparator>
 typename SkipList<Key,Comparator>::Node*
 SkipList<Key,Comparator>::NewNode(const Key& key, int height) {
-    char* mem = arena_->AllocateAligned(
+    char* mem = (char*)_mem_pool->allocate(
```
Comment (imay) [inline be/src/olap/skiplist.h]: Is this OK if we allocate not aligned?

Comment (kangkaisen) [reply be/src/olap/skiplist.h]: If I don't think wrong, _mem_pool->allocate is equal with arena_->AllocateAligned.
```
    /// Allocates a section of memory of 'size' bytes with DEFAULT_ALIGNMENT at the end
    /// of the the current chunk. Creates a new chunk if there aren't any chunks
    /// with enough capacity.
    uint8_t* allocate(int64_t size) {
        return allocate<false>(size, DEFAULT_ALIGNMENT);
    }
```

**Review by imay — APPROVED** [review]
LGMT

**Pull request merged**